File: web/src/api/downloadClient.ts
import { api } from './client'
import type { DownloadClient } from '../types'

export interface DownloadClientCreateParams {
  name: string
  type: 'qbittorrent' | 'transmission' | 'aria2'
  host: string
  username?: string
  password?: string
  is_default?: boolean
  extra?: Record<string, string>
}

export interface DownloadClientUpdateParams {
  name?: string
  type?: 'qbittorrent' | 'transmission' | 'aria2'
  host?: string
  username?: string
  password?: string
  is_default?: boolean
  enabled?: boolean
  extra?: Record<string, string>
}

export const downloadClientAPI = {
  list: () =>
    api
      .get<{ code: number; data: DownloadClient[] }>('/download-clients')
      .then((r) => r.data.data),

  get: (id: string) =>
    api
      .get<{ code: number; data: DownloadClient }>(`/download-clients/${id}`)
      .then((r) => r.data.data),

  create: (params: DownloadClientCreateParams) =>
    api
      .post<{ code: number; data: DownloadClient }>('/download-clients', params)
      .then((r) => r.data.data),

  update: (id: string, params: DownloadClientUpdateParams) =>
    api
      .put<{ code: number; data: DownloadClient }>(`/download-clients/${id}`, params)
      .then((r) => r.data.data),

  delete: (id: string) =>
    api.delete(`/download-clients/${id}`).then((r) => r.data),

  test: (id: string) =>
    api.post(`/download-clients/${id}/test`).then((r) => r.data),
}

File: internal/model/download_client.go
// Package model 定义下载客户端配置数据模型。
package model

// DownloadClient 下载客户端配置（qBittorrent / Transmission / Aria2）。
// 支持多客户端并行运行，一个标记为默认客户端。
type DownloadClient struct {
	Base
	Name      string `gorm:"size:128;not null" json:"name"`
	Type      string `gorm:"size:32;not null" json:"type"`       // qbittorrent / transmission / aria2
	Host      string `gorm:"size:512;not null" json:"host"`       // http://host:port
	Username  string `gorm:"size:256" json:"username"`
	Password  string `gorm:"size:1024" json:"-"`                 // AES加密存储
	IsDefault bool   `gorm:"default:false" json:"is_default"`
	Enabled   bool   `gorm:"default:true" json:"enabled"`
	Extra     string `gorm:"type:text" json:"-"`                 // JSON配置, AES加密
}
